### DISCUSSION TITLE
How do I use a database adapter without requiring email?

### DISCUSSION THREAD

**Discussion by zaytri** [nextauthjs/next-auth#8776] [Help]
### Summary

For context, I am trying to use NextAuth for streamer authentication for my app, specifically Twitch and YouTube. I don't need any email sign in whatsoever, because the authentication will be used to connect to Twitch and YouTube APIs to fetch stream information, such as live chat. I am trying to use the Prisma adapter for this purpose, but it seems like all database adapters are built to require email being saved in the database? For privacy reasons, I don't want email saved. It's entirely unnecessary, the only thing I need to identify a user is their `providerId`. It also seems to want emails to be unique to each user? Even though multiple Twitch accounts can have the same email.

**Comment by Naught0** [comment]
I am having to create a contrived unique email for my users as it is very possible the same email will be used for several distinct logins. It would be very helpful if there was a path to use another value to determine a user's unique identity. 

E: I assumed based on the default `false` state of `allowDangerousAccountLinking` that NextAuth would allow for this, but it does not.

**Comment by hualafun** [comment]
you can make a adapter belong to you from official adapter
